From 634116c056cff2f3df9c7fec851460e1cef62adb Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 1 Dec 2022 12:09:58 +0100
Subject: [PATCH 0388/1695] drivers: stm32_rng: add debug traces for test
 purpose

Adds:
-Dump of RNG_CR register to check clock division status and custom
NIST configuration.
-Dump of RNG_NSCR to check the noise source control configuration
-Dump of RNG_HTCR to check the health test configuration

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b8a4c79403e33f509ac77a1490249b832ae9d391 drivers: stm32_rng: add debug traces for test purpose]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I3a0c52c1a8c074fa487ebc04511627282e5fef5a
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/297213
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Lionel DEBIEVE <lionel.debieve@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/drivers/stm32_rng.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/core/drivers/stm32_rng.c b/core/drivers/stm32_rng.c
index ff4251a15..07688333e 100644
--- a/core/drivers/stm32_rng.c
+++ b/core/drivers/stm32_rng.c
@@ -282,6 +282,13 @@ static TEE_Result init_rng(void)
 		/* No need to wait for RNG_CR_CONDRST toggle as we enable clk */
 		io_clrsetbits32(rng_base + RNG_CR, RNG_CR_CONDRST,
 				RNG_CR_RNGEN);
+
+		DMSG("RNG control register %#"PRIx32,
+		     io_read32(rng_base + RNG_CR));
+		DMSG("RNG noise source control register %#"PRIx32,
+		     io_read32(rng_base + RNG_NSCR));
+		DMSG("RNG health test register %#"PRIx32,
+		     io_read32(rng_base + RNG_HTCR));
 	} else {
 		io_setbits32(rng_base + RNG_CR, RNG_CR_RNGEN | cr_ced_mask);
 	}
-- 
2.47.2

